src/commands/talk.ts

```diff
@@ -0,0 +1,224 @@
+// External dependencies
+import { writeFileSync, createReadStream } from 'node:fs';
+import { isNumber } from 'lodash';
+import {
+	SlashCommandBuilder,
+	AttachmentPayload,
+	ChannelType,
+	PermissionFlagsBits,
+	CommandInteractionOptionResolver,
+} from 'discord.js';
+import {
+	createAudioResource,
+	createAudioPlayer,
+	joinVoiceChannel,
+	getVoiceConnection,
+	AudioPlayerStatus,
+	VoiceConnectionStatus,
+	NoSubscriberBehavior,
+	entersState,
+	AudioResource,
+} from '@discordjs/voice';
+import { say, Speaker } from 'dectalk';
+import { fileSync } from 'tmp';
+
+// Internal depedencies
+import * as logger from '../logger';
+
+const builder = new SlashCommandBuilder()
+	.setName('talk')
+	.setDescription('Uses Dectalk to speak the given message')
+	.addStringOption(option =>
+		option.setRequired(true).setName('message').setDescription('The message to speak')
+	)
+	.addIntegerOption(option => {
+		option.setRequired(false).setName('speaker').setDescription('Whose voice to use');
+
+		Object.values(Speaker)
+			.filter(isNumber)
+			.forEach(value => {
+				option = option.addChoices({ name: Speaker[value] as string, value: value });
+			});
+
+		return option;
+	});
+
+export const talk: GlobalCommand = {
+	info: builder,
+	requiresGuild: false,
+	async execute(context) {
+		const options = context.interaction.options as CommandInteractionOptionResolver;
```

Comment:
This type cast is dangerous. The options resolver is defined to have the methods we use below if the interaction is a `ChatInputCommandInteraction`. I must have forgotten this distinction when creating the `CommandContext` types.

Meantime, checking that the interaction is a `ChatInputCommandInteraction` makes the requisite methods available, without spoopy typecasts:
```suggestion
		if (!context.interaction.isChatInputCommand()) throw new Error("...");
		const options = context.interaction.options;
```

Reply:
I'm working on a new PR to make this check unneeded. Slash commands should always get a `ChatInputCommandInteraction`.

Reply:
See #58

Reply:
I'll fix this after #58 is merged. Leaving this thread unresolved for now.